lib/srv/regular/sshserver.go

```diff
@@ -860,19 +866,43 @@ func (s *Server) handleSessionRequests(sconn *ssh.ServerConn, identityContext sr
 	ctx.AddCloser(ch)
 	defer ctx.Close()
 
+	// Create a close context used to signal between the server and the
+	// keep-alive loop when to close th connection (from either side).
+	closeContext, closeCancel := context.WithCancel(context.Background())
+	defer closeCancel()
+
+	clusterConfig, err := s.GetAccessPoint().GetClusterConfig()
```

Comment:
ditto for the error handling logic and cleanup

Reply:
Everything should be cleaned up in a defer here.